### PR TITLE
Permit multi-record custom field sets to extend any entity (in theory)

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Permits any entity to have multi-record custom field sets, by improving the underlying metadata.

Before
----------------------------------------
Multi-record custom sets were hardcoded in multiple places as "Contact only".
Crufty form for saving custom fieldsets.

After
----------------------------------------
Metadata exists about which entities support multi-record custom sets. This is still only enabled for contacts, but now the switch could be flipped for other entities.
Refactors the custom group form to be metadata-driven & respect the new setting (and cleanup lots of cruft). UX much nicer.

Comments
----------------------------------------
The motivation was to improve metadata, but it turned into a big cleanup project.

### CODE REVIEW

File: CRM/Custom/Form/Group.php [hunk 1]
```diff
@@ -286,77 +235,41 @@ public function buildQuickForm() {
...
-      }
-
-      if ($extends == 'Relationship' && !empty($this->_subtypes)) {
-        $relationshipDefaults = [];
-        foreach ($defaults['extends'][1] as $donCare => $rel_type_id) {
-          $relationshipDefaults[] = $rel_type_id;
-        }
-
-        $defaults['extends'][1] = $relationshipDefaults;
-      }
```
Comment (colemanw) [inline CRM/Custom/Form/Group.php:341]: This is why the PR manages to *add* functionality and metadata while not overall increasing SLOC. I deleted a bunch of code like this which:
1. Does nothing.
2. Has done nothing for over a decade.
3. Looks scary so everyone was afraid to touch it.

**Comment by seamuslee001** [comment]
@colemanw this seems to mostly work. I did find one issue which is I was just testing changing the Entity type around and if I selected Participants, then elected Role or Event name or some sub. Then changed the entity to be Contacts the extends select box still shows

contact-custom-options

**Comment by colemanw** [comment]
Thanks @seamuslee001 - fixed now.

**Comment by seamuslee001** [comment]
ok seems to work correctly now on my local merging

**Pull request merged**